pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -273,6 +276,16 @@ type AlertmanagerConfigMatcherStrategy struct {
 	Type string `json:"type,omitempty"`
 }
 
+// AlertmanagerContinueStrategy defines the strategy used by AlertmanagerConfig objects to continue the alert routing.
+type AlertmanagerContinueStrategy struct {
+	// If set to `Always`, the operator sets `continue: true` to AlertmanagerConfig object for all its routes.
```

Comment:
Do you mean by changing the AlertManagerConfig itself? It's doesn't change the AlertManagerConfig resource.

But it overrides the continue field during the configBuilder execution to generate the AlertManager JSON.  Please, CMIIW, but I think the existing behavior is to set continue to true for every route and its subroutes when using the CR AlertManagerConfig, because all the routes pass through the processRoute function.

For example, all the tests using the CR AlertManagerConfig has to set `continue: true` to pass. For example, in this test [here](https://github.com/prometheus-operator/prometheus-operator/blob/fb35b483d3c9dd6fbf204628f3125efc85cbfbd5/pkg/alertmanager/amcfg_test.go#L1209), even without setting continue: true in the test case.

Reply:
Nope "continue: true" is only added to the top-level route of the AlertmanagerConfig object.
See https://github.com/prometheus-operator/prometheus-operator/pull/6627 and https://github.com/prometheus-operator/prometheus-operator/blob/abb219ec6cd29dbfc6235ddb13ff8f1f00e08798/pkg/alertmanager/testdata/skeleton_base_CR_with_subroutes.golden

Reply:
Yup, you're right! Sorry about the confusion. OK, I will update that comment to mention only top-level routes.